flink-metrics/flink-metrics-influxdb/src/main/java/org/apache/flink/metrics/influxdb/InfluxdbReporter.java

```diff
@@ -77,10 +81,15 @@ public void open(MetricConfig config) {
 
 		this.database = database;
 		this.retentionPolicy = getString(config, RETENTION_POLICY);
+
+		int connectTimeout = getInteger(config, CONNECT_TIMEOUT);
+		int writeTimeout = getInteger(config, WRITE_TIMEOUT);
+		OkHttpClient.Builder client = new OkHttpClient.Builder().connectTimeout(connectTimeout, TimeUnit.MICROSECONDS).writeTimeout(writeTimeout, TimeUnit.MICROSECONDS);
+
 		if (username != null && password != null) {
-			influxDB = InfluxDBFactory.connect(url, username, password);
+			influxDB = InfluxDBFactory.connect(url, username, password, client, InfluxDB.ResponseFormat.JSON);
```

Comment:
There is an overloaded version of `InfluxDBFactory.connect()` method without the last argument, please use that, so that `InfluxDB.ResponseFormat.JSON` won't be exposed here.

**Side note:** `influxDB` initialization starts to grow a little bit. **Optionally**, you can move it's initialization to a dedicated private static method.

Reply:
I am running this plugin in product env, and have another feature want to try. I will edit the side note in other PR.